### PR TITLE
refactor(i18n): changed i18n middleware to plugin and fix store switc…

### PR DESCRIPTION
**Pull request by Frodigo**
…her by url issues

## Description
Fixed wrong cookie issue.

## Related Issue
#378 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/plugins/i18n.js [hunk 1]
```diff
@@ -53,27 +53,38 @@ const prepareNewCookieString = (apiState, newStoreCode) => {
...
 
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-  if (!currentStoreCode || !findLocaleBasedOnStoreCode(currentStoreCode, i18n.locales)) {
-    await setDefaultLocale(i18n);
+export default ({ app, redirect }) => {
+  let once = true;
 
-    return;
-  }
+  app.$vsf.$magento.client.interceptors.request.use(async (r) => {
```
Comment (bartoszherba) [inline packages/theme/plugins/i18n.js:60]: I think that "r" is not the best name for a variable.

File: packages/theme/plugins/i18n.js [hunk 2]
```diff
@@ -53,27 +53,38 @@ const prepareNewCookieString = (apiState, newStoreCode) => {
...
+      const apiState = app.$vsf.$magento.config.state;
+
+      apiState.setStore(i18nCurrentLocaleCode);
+      apiState.setLocale(i18nCurrentLocaleCode);
+
+      // eslint-disable-next-line no-param-reassign
+      r.headers.cookie = prepareNewCookieString(apiState, i18nCurrentLocaleCode);
+
+    }
+
```
Comment (bartoszherba) [inline packages/theme/plugins/i18n.js:86]: You can remove this empty line

File: packages/theme/plugins/i18n.js [hunk 3]
```diff
@@ -53,27 +53,38 @@ const prepareNewCookieString = (apiState, newStoreCode) => {
...
 
-export default async ({ app }) => {
-  const { i18n } = app;
-  const currentStoreCode = readStoreCookie(app);
 
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-  if (!currentStoreCode || !findLocaleBasedOnStoreCode(currentStoreCode, i18n.locales)) {
-    await setDefaultLocale(i18n);
+export default ({ app, redirect }) => {
+  let once = true;
```
Comment (bartoszherba) [inline packages/theme/plugins/i18n.js:58]: What is this variable for?

**Review by bartoszherba — CHANGES_REQUESTED** [review]
A few small issues

File: packages/theme/plugins/i18n.js (hunk 1 above)
Comment (Frodigo) [reply packages/theme/plugins/i18n.js:60]: fixed

File: packages/theme/plugins/i18n.js [hunk 4]
```diff
@@ -53,27 +53,38 @@ const prepareNewCookieString = (apiState, newStoreCode) => {
...
+
+    if (i18nCurrentLocaleCode !== localeCookie) {
+      const apiState = app.$vsf.$magento.config.state;
+
+      apiState.setStore(i18nCurrentLocaleCode);
+      apiState.setLocale(i18nCurrentLocaleCode);
+
+      // eslint-disable-next-line no-param-reassign
+      r.headers.cookie = prepareNewCookieString(apiState, i18nCurrentLocaleCode);
+
```
[the remark this answers is not in the document]
Comment (Frodigo) [reply packages/theme/plugins/i18n.js:84]: fixed

File: packages/theme/plugins/i18n.js (hunk 3 above)
Comment (Frodigo) [reply packages/theme/plugins/i18n.js:58]: removed

**Review by bloodf — APPROVED** [review]

**Review by bartoszherba — APPROVED** [review]

**Pull request merged**